core/src/main/resources/error/error-classes.json

```diff
@@ -1,319 +1,519 @@
+        ]
       },
       "TOO_MANY_TYPE_ARGUMENTS_FOR_UDF_CLASS" : {
-        "message" : [ "UDF class with <n> type arguments." ]
+        "message" : [
+          "UDF class with <n> type arguments."
+        ]
       },
       "TRANSFORM_DISTINCT_ALL" : {
-        "message" : [ "TRANSFORM with the DISTINCT/ALL clause." ]
+        "message" : [
+          "TRANSFORM with the DISTINCT/ALL clause."
+        ]
       },
       "TRANSFORM_NON_HIVE" : {
-        "message" : [ "TRANSFORM with SERDE is only supported in hive mode." ]
+        "message" : [
+          "TRANSFORM with SERDE is only supported in hive mode."
+        ]
       }
     },
     "sqlState" : "0A000"
   },
   "UNSUPPORTED_GROUPING_EXPRESSION" : {
-    "message" : [ "grouping()/grouping_id() can only be used with GroupingSets/Cube/Rollup" ]
+    "message" : [
+      "grouping()/grouping_id() can only be used with GroupingSets/Cube/Rollup"
+    ]
   },
   "UNSUPPORTED_SAVE_MODE" : {
-    "message" : [ "The save mode <saveMode> is not supported for: " ],
+    "message" : [
+      "The save mode <saveMode> is not supported for: "
+    ],
     "subClass" : {
       "EXISTENT_PATH" : {
-        "message" : [ "an existent path." ]
+        "message" : [
+          "an existent path."
+        ]
       },
       "NON_EXISTENT_PATH" : {
-        "message" : [ "a not existent path." ]
+        "message" : [
+          "a not existent path."
+        ]
       }
     }
   },
   "UNTYPED_SCALA_UDF" : {
-    "message" : [ "You're using untyped Scala UDF, which does not have the input type information. Spark may blindly pass null to the Scala closure with primitive-type argument, and the closure will see the default value of the Java type for the null argument, e.g. `udf((x: Int) => x, IntegerType)`, the result is 0 for null input. To get rid of this error, you could:\n1. use typed Scala UDF APIs(without return type parameter), e.g. `udf((x: Int) => x)`\n2. use Java UDF APIs, e.g. `udf(new UDF1[String, Integer] { override def call(s: String): Integer = s.length() }, IntegerType)`, if input types are all non primitive\n3. set \"spark.sql.legacy.allowUntypedScalaUDF\" to \"true\" and use this API with caution" ]
+    "message" : [
+      "You're using untyped Scala UDF, which does not have the input type information. Spark may blindly pass null to the Scala closure with primitive-type argument, and the closure will see the default value of the Java type for the null argument, e.g. `udf((x: Int) => x, IntegerType)`, the result is 0 for null input. ",
+      "To get rid of this error, you could: ",
+      "1. use typed Scala UDF APIs(without return type parameter), e.g. `udf((x: Int) => x)` ",
+      "2. use Java UDF APIs, e.g. `udf(new UDF1[String, Integer] { override def call(s: String): Integer = s.length() }, IntegerType)`, if input types are all non primitive ",
+      "3. set \"spark.sql.legacy.allowUntypedScalaUDF\" to \"true\" and use this API with caution"
```

Comment:
not related to this PR, but I think we should not hardcode the config name in the error message template, we should pass it in as a parameter, in case we renamed the config name in the future. cc @MaxGekk

Reply:
As much as I like to agree this causes havoc with the messages because now the config shows up as a parameter, which it really is not.
IMHO, especially for those non DBSQL, I 'd rather update the messages than write a pre-processor of sorts to patch up the JSON files.... That's what IDE's are for.

Reply:
@cloud-fan Actually, this is only the case when a config is hard-coded. I have fixes this in the PR https://github.com/apache/spark/pull/36653